types/src/row.rs

```diff
@@ -167,6 +167,18 @@ impl From<Row> for RawRow {
     }
 }
 
+impl TryFrom<RawRow> for Row {
+    type Error = Error;
+
+    fn try_from(raw: RawRow) -> std::result::Result<Self, Self::Error> {
+        let mut shares = Vec::with_capacity(raw.shares_half.len());
+        for shr in raw.shares_half {
+            shares.push(Share::try_from(shr)?);
+        }
```

Comment:
this isn't correct and again I'm not sure if we can do better than `Row::from_raw`. Row is a structure which holds the whole row from EDS, but in it's serialized form, it drops a half which needs to be reconstructed using leopard RS. However with just a plain `shares_half`, it is not possible to know which side needs to be reconstructed.